### PR TITLE
265584/sre/add workbook and deploy it

### PR DESCRIPTION
**Pull request by jenskloster**
## Description

Inspired by the Azure dashboard that already existed in the code base, we have added a workbook resource.
The workbook currently contains a single latency SLI.

## References

* [Feature 265584](https://dev.azure.com/energinet/ENDK/_workitems/edit/265584): Monitoring of stamdataupdate on Datahub (Volt)

### CODE REVIEW

File: build/infrastructure/main/workbook-charges-template.json [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+              "showBorder": false,
+              "size": "full"
+            }
+          },
+          "name": "query - 4"
+        }
+      ],
+      "isLocked": false,
+      "fallbackResourceIds": [
+        "/subscriptions/7b7ffd68-d27e-4c75-936a-2653ecc36450/resourcegroups/rg-datahub-sharedresouces-t-001/providers/microsoft.insights/components/appi-shared-sharedres-t-001"
```
Comment (HenrikSommer) [inline build/infrastructure/main/workbook-charges-template.json:198]: I guess we need to parametrize environment(?)

File: build/infrastructure/main/workbook-charges-template.json [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            "size": 3,
+            "title": "Indsendte Stamdataopdatering bliver taget imod, behandlet og tilgængelig for afhæntning med enten godkendelse eller afvisning ind for 5 minutter for 99% af beskederne",
```
Comment (HenrikSommer) [inline build/infrastructure/main/workbook-charges-template.json:149]:
```suggestion
            "title": "Indsendte Stamdataopdatering bliver taget imod, behandlet og tilgængelig for afhentning med enten godkendelse eller afvisning ind for 5 minutter for 99% af beskederne",
```

File: build/infrastructure/main/workbook-charges-template.json [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+      "type": "string",
+      "defaultValue": "${resouce_group_name}",
+      "metadata": {
+        "description": ""
+      }
+    }
+	,
+    "function": {
+      "type": "string",
+      "defaultValue": "func-functionhost-charges-u-001",
```
Comment (HenrikSommer) [inline build/infrastructure/main/workbook-charges-template.json:50]: Parametrized environment(?)

Comment (jenskloster) [reply build/infrastructure/main/workbook-charges-template.json:50]: will delete it :) it's not needed

File: build/infrastructure/main/workbook-charges-template.json [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
...0),\r\n    _time = min(timestamp)\r\n    by operation_Id\r\n| where (operations contains \"ChargeIngestion\" \r\n    and operations contains \"ChargeRejectionDataAvailableNotifierEndpoint\")\r\n    or\r\n    (operations contains \"ChargeIngestion\"\r\n    and operations contains \"ChargeConfirmationDataAvailableNotifierEndpoint\" \r\n    and operations contains \"ChargeIntegrationEventsPublisherEndpoint\" \r\n    and operations contains \"ChargeDataAvailableNotifierEndpoint\");\r\nlet history = base\r\n| make-series   sliTrend = todecimal(count()) / countif(time_taken < timespan(300s))\r\n                      on _time \r\n                from {period_dropdown:start} \r\n                to {period_dropdown:end} \r\n                step {period_dropdown:grain} \r\n| extend key=\"joinQuery\";\r\nbase\r\n| summarize \r\n    total_count = count(),\r\n    count_ok = countif(time_taken < timespan(300s))\r\n| extend sli = round(todecimal(count_ok) / total_count * 100, 2)\r\n| extend details = strcat(\"Total: \", tostring(total_count), \" OK: \", tostring(count_ok))\r\n| extend header = \"Latency\"\r\n| extend key = \"joinQuery\"\r\n| join (history) on key\r\n| project-away key1, key\r\n",
```
Comment (HenrikSommer) [inline build/infrastructure/main/workbook-charges-template.json:111]: It is quite cumbersome to read and understand the query in full when written in one line. Is it possible to separate in more lines where the `\r\n`-sign are?

File: build/infrastructure/main/workbook-charges-template.json [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+            "size": 3,
+            "title": "Indsendte Stamdataopdatering bliver taget imod, behandlet og tilgængelig for afhentning med enten godkendelse eller afvisning ind for 5 minutter for 99% af beskederne",
```
Comment (HenrikSommer) [inline build/infrastructure/main/workbook-charges-template.json:113]:
```suggestion
            "title": "Indsendte Stamdataopdatering bliver taget imod, behandlet og tilgængelig for afhentning med enten godkendelse eller afvisning inden for 5 minutter for 99% af beskederne",
```

File: build/infrastructure/main/workbook-charges.tf [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  template = file("${path.module}/workbook-charges-template.json")
+  vars = {
+    workbook_display_name     = "Charges - Stamdataopdatering SLI"
+    workbook_id               = "67d32b14-3f26-4840-a1cc-c47ca4d052de"
+    subscription_id           = data.azurerm_subscription.this.subscription_id
+    resouce_group_name        = azurerm_resource_group.this.name
+    application_insight_name  = data.azurerm_key_vault_secret.appi_shared_name.value
+    shared_resouce_group_name = var.shared_resources_resource_group_name
+
+  }
```
Comment (HenrikSommer) [inline build/infrastructure/main/workbook-charges.tf:24]:
```suggestion
    shared_resouce_group_name = var.shared_resources_resource_group_name
  }
```

File: build/infrastructure/main/workbook-charges-template.json (hunk 4 above)
Comment (jenskloster) [reply build/infrastructure/main/workbook-charges-template.json:111]: good point - We'll give it a try

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**